chapters/grid-search.qmd

```diff
@@ -0,0 +1,390 @@
+
+```{r}
+#| label: fig-bst-regular-grid-tuning
+#| echo: false
+#| out-width: 50%
+#| fig-width: 4
+#| fig-height: 3.5
+#| fig-cap: "The boosted tree tuning results for the regular grid shown in @fig-bst-grids."
+
+autoplot(bst_reg_res)  +
+  labs(y = "Brier Score") +
+  theme(legend.position = "top")
+```
+
+The pattern shown in this visualization is interesting: the trajectory for the number of trees is different for different values of the learning rate. The large two learning rates are similar, showing that the optimal value is in the mid- to low-range of our grid. The results for the smallest learning rate indicates that better performance might be found using _more_ trees than we used in our grid. This indicates that there is an _interaction effect_ in our tuning parameters (just as we saw for predictors in @sec-interactions). This might not affect how we select the best candidate value for this grid, but it does help build some intuition for this particular model that might come in handy when you tune future models. 
+
+Gaining intituion is much easier for regular grids than other designs since we have a full set of combinations. As we’ll see shortly, this interaction is very hard to see for a space-filling design. 
+
+The primary downside to regular grids is that, as the number of tuning parameters increases, the number of points required to fill the space becomes extremely large (due to the curse of dimensionality). However, for some models and pre-processing methods, regular grids can be very efficient despite the number of tuning parameters (see the following section that describes the "submodel trick").  
+
+## Irregular Grids {#sec-irregular-grid}
+
+Irregular grids are not factorial in nature. A simple example is a _random grid_ where points are randomly placed using a uniform distribution on an appropriate range for each tuning parameter. A design of size `r sfd_size` is shown in @fig-bst-grids. There are some gaps and clustering of the points, but the space for each individual dimension is covered well. The tuning results are shown in the top panels of @fig-bst-irregular-grid-tuning. Because it is an irregular design, we can't use the same visualization as in @fig-bst-regular-grid-tuning. Instead, a "marginal" plot is shown, where each numeric tuning parameter is plotted against performance in a separate panel. 
+
+The top left panel suggests that a very small learning rate is very bad, but otherwise, there is not a strong trend. For the number of trees (in the top right panel), a very small should be avoided, and perhaps the maximum number tested would be a good choice. From this plot, it is very difficult to discover the interaction effect seen with the regular grid. However, this marginal plot is the main technique for visualizing the results when there are a moderate to large number of tuning parameters. The numerically best candidate was not too dissimilar from the regular grid: `r bst_rnd_best`
```

Comment:
```suggestion
The top left panel suggests that a very small learning rate is bad, but otherwise, there is not a strong trend. For the number of trees (in the top right panel), a small number of trees should be avoided, and perhaps the maximum number tested would be a good choice. From this plot, it is very difficult to discover the interaction effect seen with the regular grid. However, this marginal plot is the main technique for visualizing the results when there are a moderate to large number of tuning parameters. The numerically best candidate was not too dissimilar from the regular grid: `r bst_rnd_best`
```

Reply:
Is it possible to discover the interaction effect with this figure?  If not, I would suggest changing "From this plot, it is very difficult to..." to "From this plot, it is impossible to..."